pwa/src/components/applications/applicationForm.tsx

```diff
@@ -0,0 +1,233 @@
+import * as React from "react";
+import {
+  GenericInputComponent,
+  Spinner,
+  Card,
+  Alert,
+  Accordion
+} from "@conductionnl/nl-design-system/lib";
+import { isLoggedIn } from "../../services/auth";
+import { Link } from "gatsby";
+import { navigate } from "gatsby-link";
+import {
+  checkValues,
+  removeEmptyObjectValues, retrieveFormArrayAsOArray,
+} from "../utility/inputHandler";
+import {ArrayInputComponent} from "../common/arrayInput";
+import FlashMessage from 'react-flash-message';
+
+export default function ApplicationForm({ id }) {
+  const [context, setContext] = React.useState(null);
+  const [alert, setAlert] = React.useState(null);
+  const [application, setApplication] = React.useState(null);
+  const [showSpinner, setShowSpinner] = React.useState(false);
+
+  React.useEffect(() => {
+    if (typeof window !== "undefined" && context === null) {
+      setContext({
+        adminUrl: window.GATSBY_ADMIN_URL,
+        frontendUrl: window.GATSBY_FRONTEND_URL,
+      });
+    } else {
+      if (isLoggedIn()) {
+        if (id !== "new") {
+          getApplication();
+        }
+      }
+    }
+  }, [context]);
+
+  const getApplication = () => {
+    setShowSpinner(true);
+    fetch(`${context.adminUrl}/applications/${id}`, {
+      credentials: "include",
+      headers: { "Content-Type": "application/json", 'Authorization': 'Bearer ' + sessionStorage.getItem('jwt') },
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        setShowSpinner(false);
+        setApplication(data);
+      })
+      .catch((error) => {
+        setShowSpinner(false);
+        console.error("Error:", error);
+        setAlert(null);
+        setAlert({ type: 'danger', message: error.message });
+      });
```

Comment:
Al deze logica gaan we lostrekken naar een `apiService`, maar dat is out of scope voor deze pr.